src/configuration.rs

```diff
@@ -1,77 +1,80 @@
 use crate::envoy::RLA_action_specifier;
-use crate::glob::GlobPatternSet;
+use crate::policy_index::PolicyIndex;
 use serde::Deserialize;
-use std::collections::HashMap;
 
 #[derive(Deserialize, Debug, Clone)]
-pub struct Operation {
-    #[serde(default)]
-    pub paths: GlobPatternSet,
-    #[serde(default)]
-    pub methods: GlobPatternSet,
+pub struct Rule {
+    pub paths: Option<Vec<String>>,
+    pub hosts: Option<Vec<String>>,
+    pub methods: Option<Vec<String>>,
 }
 
 #[derive(Deserialize, Debug, Clone)]
-pub struct Rule {
-    pub operations: Option<Vec<Operation>>,
+pub struct Configuration {
     pub actions: Option<Vec<RLA_action_specifier>>,
 }
 
-impl Rule {
-    pub fn operations(&self) -> Option<&[Operation]> {
-        self.operations.as_deref()
-    }
-
-    pub fn actions(&self) -> Option<&[RLA_action_specifier]> {
-        self.actions.as_deref()
-    }
+#[derive(Deserialize, Debug, Clone)]
+pub struct GatewayAction {
+    pub rules: Option<Vec<Rule>>,
+    pub configurations: Option<Vec<Configuration>>,
 }
 
 #[derive(Deserialize, Debug, Clone)]
 pub struct RateLimitPolicy {
-    #[serde(default)]
-    hosts: GlobPatternSet,
-    rules: Option<Vec<Rule>>,
-    global_actions: Option<Vec<RLA_action_specifier>>,
-    upstream_cluster: Option<String>,
-    domain: Option<String>,
+    pub name: String,
+    pub rate_limit_domain: String,
+    pub upstream_cluster: String,
+    pub hostnames: Vec<String>,
+    pub gateway_actions: Vec<GatewayAction>,
```

Comment:
not really happy with the name `gateway_actions`.

Maybe `rate_limit_configurations`? Sometimes bad naming indicates bad structure design. If you cannot name samething, maybe it should not be there.

Dropping here the entire struct

```yaml
failure_mode_deny: true
rate_limit_policies:
  - name: toystore
    rate_limit_domain: toystore-app
    upstream_cluster: rate-limit-cluster
    hostnames: ["*.toystore.com"]
    gateway_actions:
      - rules:
          - paths: ["/admin/toy"]
            methods: ["GET"]
            hosts: ["pets.toystore.com"]
        configurations:
          - actions:
            - generic_key:
                descriptor_key: admin
                descriptor_value: "1"
```

Currently `gateway_actions` is a list of "configuration lists" and each configuration list is conditionally applied by the rule matching. 

I was tempted to add the `rules` inside the `configuration` object, sibling of `actions`. Influenced by Envoy's rate limiting design, I decided to go as it is now. Envoy's rate limiting design adds the configuration list (**not only one configuration**) at the route level. Check out envoy's doc at  [RouteAction](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-routeaction). Which is equivalent to say 
```yaml
rules:
  - paths: ["/some/path"]
    methods: ["GET"]
configurations:
  - actions:
    -  generic_key:
          descriptor_key: admin
          descriptor_value: "1"
```

The important bit I want to highlight is that, in envoy, for a given route, you can define a list of configurations, each configuration potentially (depending on the request context) yielding one rate limit descriptor.  So each configuration yields 0 or 1 rate limit descriptor. Thus, each route can, potentially, generate multiple descriptors. 

Equivalence table:

Kuadrant object | Envoys object
--- | --- 
GatewayAction struct | [config.route.v3.RouteAction](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-routeaction)
Configuration struct| [config.route.v3.RateLimit](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit)

So, each gateway action object (envoys RouteAction object) holds a list of configuration objects (a list of envoy's RateLimit objects). 

While envoy's design has rules implicit because RouteAction is defined at specific http route (method and path) in the virtualhost object, kuadrant needs to explicitly define this rules.  

Any input would be appreciated

Reply:
`rate_limit_configs` sounds better :)